src/main/java/com/hoodiecoder/enchantmentcore/CustomEnch.java

```diff
@@ -70,33 +67,17 @@
      * @param identifier the internal ID of the enchantment
      */
     public CustomEnch(EnchantmentHolder holder, String identifier) {
-        super(NamespacedKey.minecraft(returnValid(identifier)));
-        reservedKeys.add(getKey());
+        super(new NamespacedKey(holder.getOwnerPlugin(), identifier));
         coreID = nextID;
         nextID++;
-        intName = getKey().getKey();
         ownerPlugin = holder.getOwnerPlugin();
         holder.addEnchant(this);
     }
 
-    private static String returnValid(String str) {
-        int placeholderVal = 1;
-        String finalStr = str;
-        while (Enchantment.getByKey(NamespacedKey.minecraft(finalStr)) != null || reservedKeys.contains(NamespacedKey.minecraft(finalStr))) {
-            finalStr = str + "_" + placeholderVal;
-            placeholderVal++;
-        }
-        return finalStr;
-    }
-
```

Comment:
Checking the identifier is still necessary to make sure that a single plugin does not register the same key twice

Reply:
Just ask them not to do that.
Since the namespace is now based on the plugin name, doing this is unnecessary as we now parse the whole namespace and key.

Reply:
the NamespacedKey is now `pluginname:key`.
the `TestPlugin` registers the enchantment `test_enchant`, the NamespacedKey will be `testplugin:test_enchant`.
the `TestPluginAgain` registers the same enchantment name, the NamespacedKey will be `testpluginagain:test_enchant`.
So we don't have to check the name, the NamespacedKey of the two plugins are already different.